src/impl/apis/data_api_impl.py

```diff
@@ -400,3 +400,28 @@ async def run_query(
     )
 
     return new_result
+
+
+async def run_csv_query(
+    query_parameters: QueryParameters, pool: Pool, token_model: TokenModel
+) -> QueryResult:
+    result = await run_query(query_parameters, pool, token_model)
+    csv_rows = []
+    for date, data in result.data_by_date.items():
+        for source_region, value in data.items():
+            if type(value) is dict:
```

Comment:
Something to consider here is that either this will always be true while looping through, or it won't and the result will be garbled - in which case we should 500 out instead of returning a garbled csv.

Reply:
Not necessarily for the inner value (sorry, variable naming was a bear here); value can be either a string (in the case of locations) or a dict (in the case of flows, where it provides to_ and from_)

Reply:
Yeah but that won't (or really shouldn't) actually vary within a resultset, right? Even if it we somehow mangle together the two types of table together in a single resultset, they'll come back with common columns.

Reply:
Oh, I get you now; yes, within a result this should always be consistent.